monai/apps/auto3d/data_analyzer.py

```diff
@@ -0,0 +1,541 @@
+            _get_foreground_label: define foreground image with label > 0
+            save_yaml: save results to yaml file
+            get_case_stats: get stats for each case in dataset.datalist
+            _get_case_summary: summarize the results from each case using functions _intensity_summary, _stats_opt_summary.
+        Caller:
+            get_all_case_stats: The function iterates dataset.datalist and call get_case_stats to generate stats. Then
+                                get_case_summary is called to combine results and save_yaml is called to save results.
+
+    Custimize statistics calculation:
+        Write a custome function for indivisual case:
+            _get_your_stats(self):
+                1. processed_data = retrive processed data from self.data, if not exist, process data (like cropping)
+                2. define a dict case_stats = {'stats1': calculate(processed_data)}. Notice the data may be a list of
+                data from different modalities. So calculate(processed_data) should return a list of stats. case_status
+                will be written to the yaml file
+                3. create a dict case_stats_summary = {'stats1': summary_function}. The summary_function will process
+                a list of ['stats1','stats1',...] from all cases in self.dataset.datalist
+                4. update self.data with processed_data and update self.gather_summary to register case_stats_summary,
+                and return case_stats
+        Add _get_your_stats to self._hardcode_functions
+    """
+
+    def __init__(
+        self,
+        datalist: Union[str, Dict],
+        dataroot: str,
+        output_yaml: str = "./data_stats.yaml",
+        average: bool = False,
+        do_ccp: bool = True,
+        device: str = "cuda",
+        worker: int = 2,
+    ):
+        self.output_yaml = output_yaml
+        files, _ = datafold_read(datalist=datalist, basedir=dataroot, fold=-1)
+        ds = data.Dataset(
+            data=files,
+            transform=transforms.Compose(
+                [
+                    transforms.LoadImaged(keys=["image", "label"]),
+                    transforms.EnsureChannelFirstd(keys=["image", "label"]),  # this creates label to be (1,H,W,D)
+                    transforms.Orientationd(keys=["image", "label"], axcodes="RAS"),
+                    transforms.EnsureTyped(keys=["image", "label"], data_type="tensor"),
+                    # some dataset has onehot label size of (H,W,D,C). Only allow label index of size (1,H,W,D)
+                    transforms.Lambdad(
+                        keys="label", func=lambda x: torch.argmax(x, dim=0, keepdim=True) if x.shape[0] > 1 else x
+                    ),
+                    transforms.SqueezeDimd(keys=["label"], dim=0),  # make label (H,W,D)
+                ]
+            ),
+        )
+        self.dataset = data.DataLoader(ds, batch_size=1, shuffle=False, num_workers=worker, collate_fn=lambda x: x)
```

Comment:
I think it's to avoid the stacking in collate, sometimes collate is not robust (https://github.com/Project-MONAI/MONAI/issues/2948) and it's not really needed here. Perhaps we can use this function for readability and avoid any issues from the lambda function https://github.com/Project-MONAI/MONAI/blob/a9095d09d919e94c2c07aa74fc46c1e4f9530ecd/monai/data/utils.py#L648-L652

Reply:
Fixed in 2dc9e2e3df0a65bf0345f2be16da5a764183f7d2